src/main/java/com/uber/profiling/AgentImpl.java

```diff
@@ -53,6 +55,29 @@
     private boolean started = false;
 
     public void run(Arguments arguments, Instrumentation instrumentation, Collection<AutoCloseable> objectsToCloseOnShutdown) {
+        try {
+            ConfigProvider configProvider = arguments.getConfigProvider();
+            if (configProvider != null) {
+                Map<String, Map<String, List<String>>> extraConfig = configProvider.getConfig();
+
+                Map<String, List<String>> rootConfig = extraConfig.get("");
+                if (rootConfig != null) {
+                    arguments.updateArguments(rootConfig);
+                    logger.info("Updated arguments based on config: " + JsonUtils.serialize(rootConfig));
+                }
+                
+                if (arguments.getTag() != null && !arguments.getTag().isEmpty()) {
+                    Map<String, List<String>> overrideConfig = extraConfig.get(arguments.getTag());
+                    if (overrideConfig != null) {
+                        arguments.updateArguments(overrideConfig);
+                        logger.info("Updated arguments based on config override: " + JsonUtils.serialize(overrideConfig));
+                    }
+                }
+            }
+        } catch (Throwable ex) {
+            logger.warn("Failed to update arguments with config provider", ex);
```

Comment:
should this fail?

Reply:
Just in case configProvider.getConfig() throws exception, configProvider may be provided by user, and we do not have control on it